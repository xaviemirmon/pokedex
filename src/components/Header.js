/** @jsx jsx */
// ^^ is the pragma required for using the theme-ui sx prop on non-theme-ui elements

// Package imports
import React from "react"
import { Link } from "gatsby"
import { StaticImage } from "gatsby-plugin-image"
import { Box, jsx } from "theme-ui"

const Header = () => (
  <header data-testid="header">
    <Box
      sx={{
        maxWidth: 512,
        mx: "auto",
        px: 3,
        py: 2,
        textAlign: "center",
      }}
    >
      <Link 
        to="/"
        sx={{
          fontSize: `0px`
        }}
      >
        <StaticImage
          src="../images/Pokedex_logo.png"
          width={300}
          quality={95}
          placeholder={"tracedSVG"}
          formats={["AUTO", "WEBP", "AVIF"]}
          alt="Pokedex logo"
        />
        Pokedex home
      </Link>
    </Box>
  </header>
)

export default Header
