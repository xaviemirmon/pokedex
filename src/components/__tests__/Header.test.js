// Package imports
import React from "react"
import renderer from "react-test-renderer"

// Component imports
import Header from "../Header"

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Header />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
