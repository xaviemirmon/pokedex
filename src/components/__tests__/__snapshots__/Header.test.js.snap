// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  data-testid="header"
>
  <div
    className="css-1g1xloh-Header"
  >
    <a
      className="css-filgdf-Header"
      href="/"
    >
      Pokedex home
    </a>
  </div>
</header>
`;
